src/services/shop/shop-router.test.ts

```diff
@@ -0,0 +1,431 @@
+import { beforeEach, describe, expect, it } from "@jest/globals";
+import { delAsAttendee, getAsAttendee, postAsAttendee, postAsStaff, TESTER } from "../../common/testTools";
+import { StatusCode } from "status-code-enum";
+import Models from "../../common/models";
+import { ShopItem, ShopOrder } from "./shop-schemas";
+import { AttendeeProfile } from "../profile/profile-schemas";
+
+const TESTER_SHOP_ITEM = {
+    itemId: "test-item-1",
+    name: "Test Item",
+    price: 100,
+    isRaffle: true,
+    imageURL: "test.jpg",
+    quantity: 10,
+} satisfies ShopItem;
+
+const TESTER_SHOP_ORDER = {
+    userId: TESTER.id,
+    items: ["test-item-1"],
+    quantity: [2],
+} satisfies ShopOrder;
+
+const TESTER_PROFILE = {
+    userId: TESTER.id,
+    displayName: TESTER.name,
+    avatarUrl: TESTER.avatarUrl,
+    discordTag: TESTER.discordTag,
+    points: 1000,
+    foodWave: 1,
+} satisfies AttendeeProfile;
+
+// Initialize test data before each test
+beforeEach(async () => {
+    // Clean up any existing data
+    await Models.ShopItem.deleteMany({});
+    await Models.ShopOrder.deleteMany({});
+    await Models.AttendeeProfile.deleteMany({});
+
+    // Create fresh test data
+    await Models.ShopItem.create(TESTER_SHOP_ITEM);
+    await Models.ShopOrder.create(TESTER_SHOP_ORDER);
+    await Models.AttendeeProfile.create(TESTER_PROFILE);
+});
+
+describe("POST /shop/cart/redeem", () => {
+    it("allows staff to successfully redeem an order", async () => {
+        const response = await postAsStaff("/shop/cart/redeem")
+            .send({ userId: TESTER_PROFILE.userId })
+            .expect(StatusCode.SuccessOK);
+
+        expect(JSON.parse(response.text)).toMatchObject(TESTER_SHOP_ORDER);
+
+        // Verify inventory was updated
+        const updatedItem = await Models.ShopItem.findOne({ itemId: TESTER_SHOP_ITEM.itemId });
+        expect(updatedItem?.quantity).toBe(8);
+
+        // Verify points were deducted
+        const updatedProfile = await Models.AttendeeProfile.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedProfile?.points).toBe(800);
+
+        // Verify order was deleted
+        const deletedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(deletedOrder).toBeNull();
+    });
+
+    it("returns NotFound for non-existent order", async () => {
+        await postAsStaff("/shop/cart/redeem").send({ userId: "non-existent-user" }).expect(StatusCode.ServerErrorInternal);
+    });
```

Comment:
This test is testing for an internal server error - we likely should have that error.